### ISSUE TITLE
Lint nested test_that() usage

### ISSUE THREAD

**Issue by MichaelChirico** [jimhester/lintr#3028]
I'm not sure there's a valid use case for this, and it makes some meta-tooling more difficult if we have to account for the possibility:

```r
test_that("outer call", {
  ...
  test_that("inner_call", {
  })
  ...
})
```

My memory is maybe this is allowed for something like {patrick} which is constructing `test_that()` calls, for full flexibility? If there are no good use cases for it, we should just lint it.
